src/buffer/out/textBufferCellIterator.cpp

```diff
@@ -98,12 +118,26 @@ TextBufferCellIterator& TextBufferCellIterator::operator+=(const ptrdiff_t& move
     auto newPos = _pos;
     while (move > 0 && !_exceeded)
     {
-        _exceeded = !_bounds.IncrementInBounds(newPos);
+        // If we have an endPos, check if we've exceeded it
+        if (_endPosInclusive.has_value())

```

Comment:
i'm worried about us having two end conditions

Reply:
I'll add a ton of tests to make sure this is solid 😊